src/metabase/public_settings.clj

```diff
@@ -92,13 +92,27 @@
   :type    :boolean
   :default false)
 
+(def ^:private ^:const global-max-caching-kb
```

Comment:
This is a somewhat arbitrary 200MB. I'm not sure what the right value is here. Each database supports a slight different maximum size. H2 (2GB), PostgreSQL (1GB), MySQL (4GB). From what I have read, even though the database supports those sizes, files that large can significantly affect performance. It sounds like the MySQL/PostgreSQL datatypes we're using don't support streaming and will buffer themselves in memory. So although PostgreSQL can store 1GB, we don't want to get close to that for general performance reasons.

I put 200MB here is something that is probably bigger than what users would configured to store but doesn't seem likely to crash an instance if it was held in memory.

Reply:
200MB is a reasonable approximation of "infinity" in terms of results that should be cached, i.e. much higher than someone should expect to be cached. 

👍